perf(frontend): lazy load the unified search modal

st3iny opened the pull request:
* Resolves: # <!-- related github issue -->

## Summary

Defer loading the unified search modal until it is first opened. Show a loading spinner while the modal is being loaded in the background.

(Bandwidth has been throttled during the recording to exaggerate the effect.)

https://github.com/nextcloud/server/assets/1479486/4d3aa98e-f0f8-4ed5-bf54-10199bc5f7be

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

st3iny committed:
perf(frontend): lazy load the unified search modal

Signed-off-by: Richard Steinmetz <richard@steinmetz.cloud>

ShGKme commented on core/src/views/UnifiedSearch.vue:
```diff
@@ -48,13 +53,15 @@ export default {
 	},
 	methods: {
-		toggleUnifiedSearch() {
+		async toggleUnifiedSearch() {
+			this.loadSearchModal = true
 			this.showUnifiedSearch = !this.showUnifiedSearch
 		},
```
Why is this method `async`?

st3iny replied:
Done

ShGKme commented on core/src/views/UnifiedSearch.vue:
```diff
@@ -29,14 +29,19 @@
 			</template>
 		</NcButton>
-		<UnifiedSearchModal :is-visible="showUnifiedSearch" @update:isVisible="handleModalVisibilityChange" />
+		<UnifiedSearchModal v-if="loadSearchModal"
+			:is-visible="showUnifiedSearch"
+			@hook:mounted="showUnifiedSearch = true"
+			@update:isVisible="handleModalVisibilityChange" />
```
Is `@hook:mounted` needed? `showUnifiedSearch` seems to be `true` already on the moment of mounted, because it was set in `toggleUnifiedSearch` before `UnifiedSearchModal` was top be rendered

st3iny replied:
Done

nfebe reviewed (commented):
Test and it works, please address earlier comments.

susnux reviewed (commented):
I would rather recommend not lazy load the unified search modal but show a modal with NcEmptyContent and loading spinner and just lazy load the modal content.

st3iny commented:
I implemented a loading spinner and added a screencast to the description. Note that I throttled my bandwidth during the recording to exaggerate the effect. It loads **much** faster on a regular connection.

nfebe reviewed (approved):
Not tested, code looks okay

st3iny commented:
I'm closing this one as the benefit is too small.

![grafik](https://github.com/nextcloud/server/assets/1479486/802ca1b4-9b50-467e-8860-273fab7b8a31)
